### PR TITLE
Allow uppercase chars in service and plan names

### LINKED ISSUE #422: Mixed case Plans
Right now the spec requires that plan names be all lowercase letters, however CF allows mixed-case letters. While we could look at this as bug in CF, since CF is the baseline implementation on which the spec is based (and we don't want to break existing brokers), I'm wondering if we need to loosen the language in the spec or ask CF to be spec compliant.  I'm leaning towards changing the spec.

### PR DESCRIPTION
**Pull request by duglin**
Closes #422

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by angarg12** [comment]
LGTM

File: spec.md [hunk 1]
```diff
@@ -353,7 +359,7 @@ It is therefore RECOMMENDED that implementations avoid such strings.
 
 | Response field | Type | Description |
 | --- | --- | --- |
-| name* | string | A CLI-friendly name of the service. MUST only contain lowercase characters, numbers and hyphens (no spaces). MUST be unique across all service objects returned in this response. MUST be a non-empty string. |
+| name* | string | A CLI-friendly name of the service. MUST only contain alphanumerica characters and hyphens (no spaces). MUST be unique across all service objects returned in this response. MUST be a non-empty string. |
```
Comment (jwforres) [inline spec.md]: typo? alphanumeric

**Comment by mattmcneeney** [comment]
LGTM

File: spec.md (hunk 1 above)
Comment (duglin) [reply spec.md]: fixed!

**Comment by duglin** [comment]
fixed typo and added text around how names are now case-sensitive

**Comment by duglin** [comment]
re-reviews needed due to minor tweaks

**Comment by mattmcneeney** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by zrob** [comment]
lgtm

**Pull request merged**